### PR TITLE
[Harjun751] iP

### PR DESCRIPTION
**Pull request by Harjun751**
# Marvin

> Why should I want to make anything up? Life's bad enough as it is without wanting to invent any more of it. - Douglas Adam ([source](https://www.goodreads.com/quotes/309185-why-should-i-want-to-make-anything-up-life-s-bad))

Marvin understands the limitations of human memory. He apathetically helps you remember your lists! He's:
- text-based
- depressed
- easy to use
- ~difficult~ simple to manage :smile:

All you need to do is:
1. Download it [here](https://downloadmoreram.com/)
2. Double-click it
3. Add your tasks
4. Try to contain your excitement

And it is *FREEEEEEEE!!!!!!!!!!!*

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here's the exciting `main` method:

```java
    public static void main(String[] args) {
        new Marvin().run();
    }
```

### CODE REVIEW

File: src/main/java/marvin/command/DeleteTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public DeleteTaskCommand(int index) {
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList taskList) {
+        try {
+            String old = taskList.removeTask(index);
+            Ui.printGeneric("I've removed the task.\n" + old + "\nNow you have " + taskList.getCount()
```
Comment (junwei07) [inline src/main/java/marvin/command/DeleteTaskCommand.java]: You can keep the line length shorter.

File: src/main/java/marvin/command/Command.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package marvin.command;
+
+import marvin.task.TaskList;
+
+public abstract class Command {
```
Comment (junwei07) [inline src/main/java/marvin/command/Command.java:8]: You can include a header comment for this class.

File: src/main/java/marvin/task/Task.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        this.done = done;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public boolean getDone() {
+        return this.done;
+    }
```
Comment (junwei07) [inline src/main/java/marvin/task/Task.java]: You can add header comments for these methods.

File: src/test/java/marvin/ParserTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Test
+    public void parse_invalidMarkInput_throwsException(){
+        try {
+            Command c = parse("mark abc");
+            fail();
+        } catch (Exception e) {
+            assertEquals(MarvinException.class, e.getClass());
+        }
+    }
```
Comment (junwei07) [inline src/test/java/marvin/ParserTest.java]: You can include another edge cases, e.g. when the input is "mark".

**Review by junwei07 — COMMENTED** [review]
LGTM, overall good structure of code.
